components/locale_core/src/data.rs

```diff
@@ -386,12 +423,16 @@ fn test_data_locale_to_string() {
         },
         TestCase {
             locale: "und-u-sd-sdd",
-            expected: "und-u-sd-sdd",
+            expected: "und-SD-u-sd-sdd",
```

Comment:
Question: I'm not seeing where the region subtag would get auto-populated during parsing or stringification? Does it go through LocalePreferences now?

Reply:
because we have public fields, it needs to be done during stringification. it happens in `region_and_subdivision` which is also used for `PartialEq`/`Hash`.